sky/serve/core.py

```diff
@@ -28,6 +28,67 @@
 if typing.TYPE_CHECKING:
     from sky import clouds
 
+logger = sky_logging.init_logger(__name__)
+
+
+def _validate_service_task(task: 'sky.Task'):
+    """Validate the task for Sky Serve.
+
+    Args:
+        task: sky.Task to validate
+    """
+    spot_resources: List['sky.Resources'] = [
+        resource for resource in task.resources if resource.use_spot
+    ]
+    # TODO(MaoZiming): Allow mixed on-demand and spot specification in resources
+    # On-demand fallback should go to the resources specified as on-demand.
+    if len(spot_resources) not in [0, len(task.resources)]:
+        with ux_utils.print_exception_no_traceback():
+            raise ValueError(
+                'Resources must either all use spot or none use spot. '
+                'To use on-demand and spot instances together, '
+                'use `dynamic_ondemand_fallback` or set '
+                'base_ondemand_fallback_replicas.')
+
+    if task.service is None:
+        with ux_utils.print_exception_no_traceback():
+            raise RuntimeError('Service section not found.')
+
+    for resource in list(task.resources):
+        if resource.spot_recovery is not None:
+            with ux_utils.print_exception_no_traceback():
+                raise ValueError('spot_recovery is disabled for SkyServe. '
+                                 'SkyServe will replenish preempted spot.')
```

Comment:
```suggestion
                                 'SkyServe will replenish preempted spot with... ')
```
add desc for our policy